common/source/core/Registry.hpp

```diff
@@ -66,19 +68,63 @@ class Registry : public ISerializable {
 			return m_recipes.back().get();
 		}
 
+		template<typename T>
+		auto registerTree(const std::string &stringID, const std::string &label) -> typename std::enable_if<std::is_base_of<Tree, T>::value, T&>::type {
+			size_t id = m_trees.size();
+			m_treesID.emplace(stringID, id);
+			m_trees.emplace_back(id, stringID, label);
+			return m_trees.back();
+		}
+
+		template<typename T>
+		auto registerSerializedTree(sf::Packet &packet) -> typename std::enable_if<std::is_base_of<Tree, T>::value, T&>::type {
+			m_trees.emplace_back();
+			m_trees.back().deserialize(packet);
+
+			size_t id = m_trees.size() - 1;
+			m_treesID.emplace(m_trees.back().stringID(), id);
+
+			return m_trees.back();
+		}
+
+		template<typename T>
+		auto registerBiome(const std::string &stringID, const std::string &label) -> typename std::enable_if<std::is_base_of<Biome, T>::value, T&>::type {
+			size_t id = m_biomes.size();
+			m_biomesID.emplace(stringID, id);
+			m_biomes.emplace_back(id, stringID, label);
+			return m_biomes.back();
+		}
+
+		template<typename T>
+		auto registerSerializedBiome(sf::Packet &packet) -> typename std::enable_if<std::is_base_of<Biome, T>::value, T&>::type {
```

Comment:
`registerTree`, `registerSerializedTree`, `registerBiome` and `registerSerializedBiome` don't need to be templates since T is never actually used in the code.

Please make them normal functions and define their code in the C++ (just like `registerItem` and `registerSerializedItem`).